x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/jwt/JwtRealm.java

```diff
@@ -527,4 +675,81 @@ static BytesKey computeBytesKey(final CharSequence charSequence) {
         messageDigest.update(charSequence.toString().getBytes(StandardCharsets.UTF_8));
         return new BytesKey(messageDigest.digest());
     }
+
+    private boolean reloadJwksHelper(ActionListener<AuthenticationResult<User>> listener, String tokenPrincipal, boolean isJwtAlgHmac) {
```

Comment:
I think we should try avoid having method that both calls listener and returns a value. There may be exceptions. But it often adds to confusion and is prone to bugs.

Reply:
I will refactor the method to remove the return value.